### PR TITLE
[Examples] refactor examples and add github action to automatically validate the scripts.

### PR DESCRIPTION
**Pull request by TheaperDeng**
### 1. Motivation and Context
Our example collection is kind of messy and hard to navigate. This PR aims to make the organized and add github action to test them to enable long-term usability.

**This PR does not add any examples. New example scripts will be updated according to the necessity and request.**

### 2. Summary of the change
1. Rename the folder `/example` -> `/examples`
2. Reorganize current examples into three catagories (noisy label detection, use pretrained benchmark, and estimate brittleness). See [here](https://github.com/TRAIS-Lab/dattri/blob/586c344ff7f5105eac6d1f6f2fe49c0853eafaaf/examples/readme.md) for a detailed guide. 
3. Fix all 6 examples to ensure that both CPU and GPU users could run through them in a reasonable time (<2min), delete some redundant examples.
4. Add a github action to run the examples for each PR, so that broken examples will be found out timely. The github action will only check the completion of example running. Accuracy monitoring will be added in later PRs.
5. Some minor changes in the `dattri` repo (add cifar10 support, `AttributionTask` will load the checkpoints in the same device as the model, auc will transform the score to cpu to avoid user redundant code).

### 3. What tests have been added/updated for the change?
- [x] Unit test: Typically, this should be included if you implemented a new function/fixed a bug.
- [x] Application test: If you wrote an example for the toolkit, this test should be added.

### CODE REVIEW

File: examples/noisy_label_detection/influence_function_noisy_label.py [hunk 1]
```diff
@@ -27,43 +27,35 @@
...
-
-
 if __name__ == "__main__":
-
     parser = argparse.ArgumentParser()
     parser.add_argument("--method", type=str, default="explicit")
     parser.add_argument("--device", type=str, default="cuda")
     args = parser.parse_args()
 
+    # load the dataset, we only need the train dataset
```
Comment (jiaqima) [inline examples/noisy_label_detection/influence_function_noisy_label.py:36]: nit: this line is not grammatically correct

File: examples/noisy_label_detection/tracin_noisy_label.py [hunk 2]
```diff
@@ -1,73 +1,72 @@
...
-    dataset.targets, flip_index = flip_label(dataset.targets[subset_indice], p=0.1)
-    return flip_index
-
-
 if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--device", type=str, default="cuda")
+    args = parser.parse_args()
+
+    # load the dataset, we only need the train dataset
```
Comment (jiaqima) [inline examples/noisy_label_detection/tracin_noisy_label.py:19]: same grammatical issue

File: examples/noisy_label_detection/trak_noisy_label.py [hunk 3]
```diff
@@ -1,36 +1,43 @@
...
+    )
     return flip_index
 
 
 if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--device", type=str, default="cuda")
+    args = parser.parse_args()
+
+    # create cifar 10 dataset
```
Comment (jiaqima) [inline examples/noisy_label_detection/trak_noisy_label.py:28]: cifar 10 or cifar 2?

File: examples/noisy_label_detection/trak_noisy_label.py [hunk 4]
```diff
@@ -42,17 +49,18 @@ def get_mnist_indices_and_adjust_labels(dataset):
         sampler=SubsetSampler(range(1000)),
     )
 
-    model = train_mnist_lr(train_loader_full)
-    model.cuda()
+    model = train_cifar2_resnet9(train_loader, num_epochs=3, num_classes=10)
```
Comment (jiaqima) [inline examples/noisy_label_detection/trak_noisy_label.py:52]: maybe rename this function as train_cifar_resnet9?

File: examples/readme.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
+# `dattri` examples
+This folder contains bite-sized examples which can help users to build their own application by `dattri`.
```
Comment (jiaqima) [inline examples/readme.md:2]: This folder contains bite-sized examples that can help users build their own applications with `dattri`.

File: examples/readme.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+## Noisy label detection
+This section contains using different attributors to detect the noisy label in various datasets.
```
Comment (jiaqima) [inline examples/readme.md:5]: This section contains examples using different attributors to detect noisy labels in various datasets.

File: examples/readme.md [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+[Use influence function to detect noisy label in Mnist10 + Logistic regression.](./noisy_label_detection/influence_function_noisy_label.py)
```
Comment (jiaqima) [inline examples/readme.md:7]: label -> labels

File: examples/readme.md [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+
+[Use TracIN to detect noisy label in Mnist10 + MLP.](./noisy_label_detection/tracin_noisy_label.py)
```
Comment (jiaqima) [inline examples/readme.md:9]: label -> labels

File: examples/readme.md [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+
+[Use TRAK to detect noisy label in CIFAR10 + ResNet-9.](./noisy_label_detection/trak_noisy_label.py)
```
Comment (jiaqima) [inline examples/readme.md:11]: label -> labels

File: examples/readme.md [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+
+## Use pretrained checkpoints and pre-calculated ground truth
+
+This section contains examples to use the pretrained checkpoints and pre-calculated ground truth provided by `dattri` to evaluate the data attribution methods.
```
Comment (jiaqima) [inline examples/readme.md:15]: to use -> using

File: examples/readme.md [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+
+[Use pre-trained Mnist10 + MLP benchmark setting and evaluate Influence Function (CG) algorithm by LDS](./pretrained_benchmark/influence_function_lds.py)
+
+[Use pre-trained Mnist10 + LR benchmark setting and evaluate TRAK algorithm by LOO correlation](./pretrained_benchmark/trak_lds.py)
+
+## Estimate the brittleness
+
+This section contains examples to use attribution score to estimate the brittleness of a model.
```
Comment (jiaqima) [inline examples/readme.md:23]: to use -> using

attribution score -> attribution scores

**Review by jiaqima — APPROVED** [review]
Left some minor comments. Otherwise LGTM

File: examples/noisy_label_detection/trak_noisy_label.py (hunk 3 above)
Comment (TheaperDeng) [reply examples/noisy_label_detection/trak_noisy_label.py:28]: it's cifar10 because we find that cifar2 noisy label detection is far more difficult than cifar10. That's actually an interesting observation.

File: examples/noisy_label_detection/trak_noisy_label.py (hunk 4 above)
Comment (TheaperDeng) [reply examples/noisy_label_detection/trak_noisy_label.py]: good catch. I have revised the module to be named as `dattri.benchmark.datasets.cifar` that contains both cifar2 and cifar10 functions.
